### DISCUSSION TITLE
Creating a custom AuthenticationStateProvider for Blazor web assembly

### DISCUSSION THREAD

**Discussion by liorbalmas** [abpframework/abp#7545] [Q&A] [answered]
Hello, 
I'm trying to use  and authentication method other then provided by the template on client side - Blazor WebAssembly.
For that, I have a implemented a custom AuthenticationStateProvider.
But that seems not enough as the framework expects implementation of more services such as IAccessTokenProvider and SignOutSessionStateManager which I could not figure how and where to implement.
It seems there's no one place where all the authentication/authorization services are created..
Thanks for any help on that.

**Comment by maliming** [comment, marked as answer]
hi
Please create an issue.
https://github.com/abpframework/abp/issues/new

**Reply by liorbalmas** [reply]
Done 
https://github.com/abpframework/abp/issues/7554#issue-798018699
Thanks
